File: mobile/lib/data/models/purchase/purchase.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'purchase.freezed.dart';
part 'purchase.g.dart';

@freezed
class Purchase with _$Purchase {
  const factory Purchase({
    required String userId,
    required double amount,
    required bool firstPurchase,
  }) = _Purchase;

  factory Purchase.fromJson(Map<String, dynamic> json) =>
      _$PurchaseFromJson(json);
}

File: api/models/purchase.js
const mongoose = require('mongoose');

const PurchaseSchema = new mongoose.Schema({
    userId: { type: String, required: true },
    amount: { type: Number, required: true },
    firstPurchase: { type: Boolean, default: false },
});

module.exports = mongoose.model('Purchase', PurchaseSchema);
